src/chapters/shell-intro/walkthrough_bash-basics.rst

```diff
@@ -0,0 +1,1043 @@
+   
+   # correct: no spaces
+   variable_name=value
+
+   # wrong: spaces between declaration and assignment
+   variable_name = value
+
+When the value of a variable does not have any spaces it can be written as shown above. When you need to have spaces you can put single-quotes (`''`) around the value. These serve to group the whole string value together including the spaces:
+
+.. sourcecode:: bash
+   :caption: Linux/BASH
+
+   variable_name='a longer value'
+
+Substituting variables
+^^^^^^^^^^^^^^^^^^^^^^
+
+Once a variable has been defined (either in the script or a global environment variable) it can be referenced by prefixing a ``$`` symbol before it:
+
+.. sourcecode:: bash
+   :caption: Linux/BASH
+
+   # define the variable
+   my_variable='hello world'
+
+   # reference the script variable
+   $my_variable
+
+   # reference a global variable
+   $PATH
+
+Variable Substitution
+^^^^^^^^^^^^^^^^^^^^^
+
+Referencing a variable is straightforward. But in most cases this process is done *inside* of a command, referencing in the open as we have done above has no effect. For this behavior BASH has a mechanism called **variable substitution**.
+
+For example, consider this simple script that creates and moves a directory using variables to hold the paths. Above each command is a comment showing what the command looks like when its variables have been substituted.
+
+Create a directory called ``bash-examples`` in your home (``~``) directory and open a new file called ``variable-substitution.sh`` within it. You can use any editor you would like to paste in the contents below.
+
+.. sourcecode:: bash
+   :caption:  ~/bash-examples/variable-substitution.sh
+
+   target_path=/tmp/dir-name
+
+   # ~ is a shorthand for /home/<username>
+   destination_path=~/dir-name
+
+   # mkdir /tmp/dirname
+   mkdir "$target_path"
+
+   # mv /tmp/dir-name ~/dir-name
+   mv "$target_path" "$destination_path"
+
+   # $HOME is an environment variable with a value of /home/<username>
+   # ls /home/<username>
+   ls "$HOME" 
+
+You should now have a file with the path ``~/bash-examples/variable-substitution.sh`` that you can execute using ``bash`` as the interpreter:
+
+.. sourcecode:: bash
+   :caption: Linux/BASH
+
+   $ bash ~/bash-examples/variable-substitution.sh
+
+You likely noticed that the variables are contained in double-quotes (``""``) when used in commands. This is a best practice when working with substitutions as it can prevent unintended behavior caused by spaces or special characters. This is especially true when scripts receive user input which, as you now know, should never be trusted!
+
+.. admonition:: note
+
+   You can read more about the behavior of `escape characters and single and double quotes <https://linux.101hacks.com/bash-scripting/quotes-inside-shell-script/>`_ in this article. If it goes over your head it's okay, just follow the best practice to stay safe and come back to understanding the *why* later.
+
+Command Substitution
```

Comment:
This is important because they see it in our class. However, I think this section is still a little deeper than students need right now.